test/parallel/test-assert.js

```diff
@@ -378,7 +378,7 @@ try {
 assert.strictEqual(true, threw,
                    'a.doesNotThrow is not catching type matching errors');
 
-assert.throws(function() { assert.ifError(new Error('test error')); });
+assert.throws(function() { assert.ifError(new Error('test error')), /^([A-Za-z])\w+$/});
```

Comment:
This regexp is probably too general. Since the error being thrown here is known, can you change that to `/^Error: test error$/`?